### PR TITLE
ROX-11335: Move Scanner DB secret out of main Scanner DB container

### PR DESCRIPTION
**Pull request by RTann**
This PR makes a few PostgreSQL-related updates:

* Move Scanner DB's secret out of the main Scanner DB container
    * This moves the secret into the init container, as it is only needed there
* Some Dockerfile cleanup

We also start looking at initContainer logs

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -652,7 +652,9 @@ jobs:
   create-postgres-dump-from-genesis-dump:
     docker:
       - *defaultImage
-      - image: postgres:12.0-alpine
+      - image: postgres:12-alpine
+        environment:
+          POSTGRES_PASSWORD: password
```
Comment (RTann) [inline .circleci/config.yml:657]: Took this from the stackrox repo, but we should probably not expose this nor make it obvious as maybe it's possible for someone who has access to the CI container to mess with the data. Any thoughts @gavin-stackrox ?

Comment (RTann) [reply .circleci/config.yml:657]: Note: we don't actually use this password for our needs. It was just required to set this environment variable in the latest postgres 12 version

**Comment by RTann** [comment]
Removing reviewers for now, as it is not as ready as i originally thought

**Review by connorgorman — APPROVED** [review]
I would prefer we unify the Central and Scanner scripts and YAMLs. Does the way that Central-db do initialization work for scanner db?

**Review by connorgorman — CHANGES_REQUESTED** [review]
sorry meant to request changes and not approve

**Comment by RTann** [comment]
> I would prefer we unify the Central and Scanner scripts and YAMLs. Does the way that Central-db do initialization work for scanner db?

Central does not initialize the database with data like Scanner does/needs: https://github.com/stackrox/stackrox/blob/3.70.0/image/templates/helm/stackrox-central/templates/01-central-12-central-db.yaml#L43. Central solely calls `docker-entrypoint.sh initdb ...` which just does `initdb` https://github.com/stackrox/stackrox/blob/3.70.0/image/postgres/scripts/docker-entrypoint.sh#L341. This just initializes the directory. We need to ensure the initial data is populated https://github.com/stackrox/scanner/blob/master/image/db/rhel/scripts/docker-entrypoint.sh#L336

**Review by connorgorman — APPROVED** [review]

**Pull request merged**